OneDrivePhotoBrowser/AccountSelection.xaml.cs

```diff
@@ -51,28 +64,33 @@ public AccountSelection()
 
         private async void AccountSelection_Loaded(object sender, RoutedEventArgs e)
         {
-            if (((App)Application.Current).OneDriveClient != null)
+            var app = ((App) Application.Current);
+            if (app.OneDriveClient != null)
             {
-                await ((App)Application.Current).OneDriveClient.SignOutAsync();
-
-                var client = ((App)Application.Current).OneDriveClient as OneDriveClient;
-                if (client != null)
+                var msaAuthProvider = app.AuthProvider as MsaAuthenticationProvider;
+                var adalAuthProvider = app.AuthProvider as AdalAuthenticationProvider;
```

Comment:
We should add an interface to onedrive-sdk-dotnet-msa-auth-adapter that both providers implement and then we just need a cast to IAuthenticationProviderWithSignOut and we are done.

```C#
public interface IAuthenticationProviderWithSignOut : IAuthenticationProvider
{
    Task SignOutAsync()
}
```

---
In reply to: [73744044](https://github.com/OneDrive/onedrive-sample-photobrowser-uwp/pull/1#discussion_r73744044) [](ancestors = 73744044,73741840)